lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -202,15 +202,17 @@ sub _setup_websocket_connection {
                     # Subprocesses reset the event loop (which triggers this event),
                     # and since only the main worker process uses the WebSocket we
                     # can safely ignore this (and do not want to reconnect)
+                    # uncoverable statement
                     return unless $websocket_pid == $$;
 
                     # ignore if the connection was disabled from our side
+                    # uncoverable statement
                     return log_debug("Websocket connection to $websocket_url finished from our side.")
                       unless $self->websocket_connection;
 
-                    $reason //= 'no reason';
-                    $self->websocket_connection(undef);
-                    $self->_set_status(
+                    $reason //= 'no reason';               # uncoverable statement
```

Comment:
At least for this one i would just change the `@_` unrolling.
```
my ($self, $code, $reason) = (shift, shift, shift // 'no reason');
```

Reply:
ok, done in https://github.com/os-autoinst/openQA/pull/3718/commits/bc4dc4aa57ce3f834bd2de1f4edaffa2d44078ce included in this PR

Reply:
I assume

```perl
finish => sub ($tx, $code, $reason = 'no reason') {
...
```

is not yet part of any stable language definition? ;)

Reply:
Actually, that's fine too, and works exactly the way you expect it to. 😁

Reply:
Cool. As tests failed anyway I did that now. Let's see :) Only that my vim syntax checker does not like it yet :D

Reply:
You have to activate signatures in the namespace (`use Mojo::Base 'Mojo::EventEmitter', -signatures;`). But then it should be fine.

Reply:
doesn't help vim syntax highlight but should solve the compile problem ;)